### PR TITLE
Wayland colorpicking

### LINKED ISSUE #598: Color picker not working under Wayland
## Expected Behavior
When clicking in the **global color picker**, it should zoom in and let me choose a color.
Like #432 gif

## Current Behavior
The color picker circle does nothing and the cursor gets stuck
Log:
```
(Akira:2): Gdk-CRITICAL **: 16:44:35.500: gdk_monitor_get_geometry: assertion 'GDK_IS_MONITOR (monitor)' failed
```

## Possible Solution
1. Use xdg portals
2. Make the color picker only local    <--- I will try to do this
3.  Delete the color picker 

## Steps to Reproduce (for bugs)
1. Run akira under Wayland 
2. Click in the color picker
3. Try to pick a color from the screen

## Context
I was trying the app features and this happened

## Your Environment (for bugs)
- Akira Version: v0.0.14 - alpha
- OS Info:
Linux pKnw 5.11.22_1 #1 SMP 1621481187 x86_64 GNU/Linux
NAME="void"
ID="void"
DISTRIB_ID="void"
PRETTY_NAME="void"

### PR DESCRIPTION
**Pull request by madjesc**
## Summary / How this PR fixes the problem?
Implementing colorpicking in Wayland using libportal

## Screenshots 
image
image

## Known Issues / Things To Do

* [ ] Handle pick_color_finish exception
* [x] Test in Xorg

## This PR fixes/implements the following **bugs/features**:
 

- Fixes #598

### CODE REVIEW

**Comment by Mkefs** [comment]
It not works with the master update, fixing

File: build-aux/flatpak/com.github.akiraux.akira.json [hunk 1]
```diff
@@ -23,7 +23,16 @@
...
     "--filesystem=xdg-run/gvfsd"
   ],
-  "modules": [{
+  "modules": [
+   {
+      "name": "libportal",
+      "buildsystem": "meson",
+      "sources": [{
+        "type": "git",
+        "url": "https://github.com/flatpak/libportal.git"
```
Comment (bilelmoussaoui) [inline build-aux/flatpak/com.github.akiraux.akira.json:32]: should ideally be a tarball of a tagged release, at worst a specific commit

File: meson.build [hunk 2]
```diff
@@ -22,6 +22,7 @@ cairo_dependency = dependency('cairo', version: '>=1.14')
 goocanvas_dependency = dependency('goocanvas-3.0')
 libarchive_dependency = dependency('libarchive')
 json_glib_dependency = dependency('json-glib-1.0')
+dep_libportal = dependency('libportal', required: true)
```
Comment (bilelmoussaoui) [inline meson.build:25]:
```suggestion
libportal_dependency = dependency('libportal')
```
the dependencies are required by default

File: build-aux/flatpak/com.github.akiraux.akira.json (hunk 1 above)
Comment (Mkefs) [reply build-aux/flatpak/com.github.akiraux.akira.json:32]: Ok.

**Comment by Mkefs** [comment]
Now it works, using libportal 4.0
And now the Utils.ColorPicker class is not a window, 
is a class that emmits a signal when the colorpicking is done.

**Comment by Alecaddd** [comment]
@Mkefs Hi, thank you so much for your contribution.
This is great, but indeed it shouldn't break this feature for X11 users, which are our primary target for now.
Any chance of making this work for both?

**Comment by Alecaddd** [comment]
Converting this to a Draft PR since it's not ready for a full review or to be merged.

**Comment by Mkefs** [comment]
It works in Xorg.
Tested on Gnome 3.38 on Debian bullseye 
imagen

**Comment by Alecaddd** [comment]
Awesome, I'll review this later today.
Thanks!

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -29,6 +29,13 @@ jobs:
           ../configure --prefix /usr
           make
           make install
+      - name: libportal-4.0 from source
+        run: |
+          git cline https://github.com/flatpak/libportal/
```
Comment (bilelmoussaoui) [inline .github/workflows/build.yml:34]:
```suggestion
          git clone https://github.com/flatpak/libportal.git
```

**Comment by Alecaddd** [comment]
The build is busted.

**Comment by Alecaddd** [comment]
@Mkefs you wrote `Handle screenshot exception` in the list of things to do.
What did you mean by that? Is that still an actionable item you want to tackle?

**Comment by Mkefs** [comment]
> @Mkefs you wrote `Handle screenshot exception` in the list of things to do.
> What did you mean by that? Is that still an actionable item you want to tackle?
libportal pick_color_finish throws a GLib.Error if something went wrong. In ColorPicker.vala I don't handle this exception thus it does nothing wen something wrong in libportal happens. Although it's harmless it should popup something to the user saying that something went wrong, or just log it to the console 

```vala 
// Libportal color picking finish 
// try {
GLib.Variant v = portal.pick_color_finish (result);
// } catch (GLib.Error e) { ... /* <-- Do something */ }
 ```

**Comment by Alecaddd** [comment]
I can deal with that, thanks.
Another question. Is `libportal` available anywhere as an installable package with apt?
I know it's kind of a silly question as we're fully transitioning to a full faltpak based build, but for those who locally build this via `meson` I'd like to avoid forcing them to manually install libportal.

**Comment by Mkefs** [comment]
> I can deal with that, thanks.
> Another question. Is `libportal` available anywhere as an installable package with apt?
> I know it's kind of a silly question as we're fully transitioning to a full faltpak based build, but for those who locally build this via `meson` I'd like to avoid forcing them to manually install libportal.

Via apt in debian no, in other distros like Void Linux, Arch and I think Ubuntu too, yes.
In debian it's in the unstable branch.

**Review by Alecaddd — CHANGES_REQUESTED** [review]
This works but it removes the fundamental feature that makes the color picker tool easy to use and superior to the default color picker. The zoom (magnifying glass) effect.
That is vital and we need to keep it.

**Comment by Mkefs** [comment]
Ok, I will try to do it using libportal

**Comment by mbfraga** [comment]
Awesome stuff! I use wayland on my dev machine that I use for Akira...so this is quite nice :P Will review.

**Comment by mbfraga** [comment]
Works on wayland (what I use now). So it definitely gets my upvote.

**Pull request closed without merging**